handlers/insolvency_resource.go

```diff
@@ -69,15 +70,33 @@ func HandleCreateInsolvencyResource(svc dao.Service) http.Handler {
 			return
 		}
 
-		// Check with company profile API if company is valid
-		err, httpStatus = service.CheckCompanyInsolvencyValid(&request, req)
+		// Check with company profile API if company exists
+		var companyProfile *companieshouseapi.CompanyProfile
+		err, httpStatus, companyProfile = service.CheckCompanyExists(&request, req)
 		if err != nil {
 			log.ErrorR(req, fmt.Errorf("company was not found valid when checking company profile API [%v]", err))
 			m := models.NewMessageResponse(fmt.Sprintf("company [%s] was not found valid for insolvency: %v", request.CompanyNumber, err))
 			utils.WriteJSONWithStatus(w, req, m, httpStatus)
 			return
 		}
 
+		//Check with alphakey service if company name valid and Check with company profile API if other details are valid
+		err, httpStatus = service.CheckCompanyNameAlphaKey(companyProfile.CompanyName, &request, req)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("company name was not found valid when checking company profile API [%v]", err))
+			m := models.NewMessageResponse(fmt.Sprintf("company name [%s] was not found valid for insolvency: %v", request.CompanyNumber, err))
+			utils.WriteJSONWithStatus(w, req, m, httpStatus)
+			return
+		} else {
+			err = service.CheckCompanyDetailsAreValid(companyProfile, &request)
```

Comment:
This check doesn't need to be within an `else` statement. The `else` statement can be removed

Reply:
amended

Reply:
amended